### DISCUSSION TITLE
Deepgram for React native (IOS, Android)

### DISCUSSION THREAD

**Discussion by hvdhavietduc** [deepgram/community#971] [General help] [answered]
Hi everyone,

I want to use Deepgram for my react native app. But I cannot use it in react native.
Can you help me ?

This is the code that I try:
export const App = () => {
 var socket = React.useRef(new WebSocket('wss://api.deepgram.com/v1/listen?model=nova-2&multichannel=true&language=en-US&channels=1', ['token', MY_TOKEN])).current;

    const keepAliveMsg = JSON.stringify({ type: "KeepAlive" });
    useEffect(() => {
        socket.onopen = () => {
            setServerState('Connected to the server')
            socket.send(keepAliveMsg);
            console.log(123123)
        };
        socket.onclose = (e) => {
            console.log(12312334)
            setServerState('Disconnected. Check internet or server.')
        };
        socket.onerror = (e) => {
            setServerState(e.type);
        };
        socket.onmessage = (e) => {
            console.log(e.data)
        };
    }, [])

    const onStart = () => {
        LiveAudioStream.init(audioOptions);
        let i = 0;
        console.log(socket)
        console.log(123)
            LiveAudioStream.on('data', (data) => {
                if(i<=0){
                    console.log(data)
                    i++;
                }
                const chunkdata = Buffer.from(data, 'base64');
                // const chunk = new Uint8Array(chunkdata);
                socket.send(chunkdata);
            });
        LiveAudioStream.start();

    };

    const onStop = async () => {
        console.log('Stream stopped:');
        const file = await LiveAudioStream.stop();
        const chunkdata = Buffer.from(file, 'base64');
        const chunk = new Uint8Array(chunkdata);
        socket.send(chunk);
        socket.close()
    };

return (
        <View className={'flex items-center w-full h-full'}>
            <Text>{serverState}</Text>
            <Button title={'Start'} onPress={onStart}/>
            <Button title={'Stop'} onPress={onStop}/>
        </View>
    );

When I try to it, I reveive the data from deepgram server:
{
  "type": "Metadata",
  "transaction_key": "deprecated",
  "request_id": "ab529f32-8ca7-4c2c-8261-4ba076638bf7",
  "sha256": "ecf68ad68e7adb19741a159fc65f7ad265a640321cae4619061eecec8516a447",
  "created": "2024-10-23T14:02:39.819Z",
  "duration": 0,
  "channels": 0
}

So can you help me ? Thank you very much!

**Comment by deepgram-community** [comment]
Hey there! It looks like you haven't connected your GitHub account to your Deepgram account. You can do this at https://community.deepgram.com - being verified through this process will allow our team to help you in a much more streamlined fashion.

**Comment by deepgram-community** [comment, marked as answer]
Hey there,

I pulled your request information, and it looks like the audio data you sent couldn't be parsed. You'll want to double check that your configuration you send to the webhook matches the format of the audio you are providing.

> ![avatar](https://avatars.slack-edge.com/2024-09-27/7801116181556_c8a1540838c25bbd27a2_32.png) This message was sent by Naomi from Deepgram, via our community automation.

**Comment by ajay-ranga** [comment]
Hi @hvdhavietduc 
How are you implementing
'LiveAudioStream'

**Comment by warlockdn** [comment]
Kindly provide an answer. This question is marked as answered but its not

**Comment by divyesh-puri** [comment]
@warlockdn Were you able to figure this out?

**Comment by MaheshtheDev** [comment]
If you guys are looking `Speech-to-Text`, this could be easiest thing to be make use native api with `@react-native-voice/voice` library. Cause i am having loads of issues with deepgram in the react native too
cc: @divyesh-puri , @warlockdn 

```typescript
import Voice, {
  type SpeechResultsEvent,
  type SpeechErrorEvent,
} from "@react-native-voice/voice";
.
.
.
  useEffect(() => {
    // Initialize Voice
    Voice.onSpeechStart = () => {
      console.log("🎤 Speech started");
      setIsRecording(true);
    };

    Voice.onSpeechEnd = () => {
      console.log("🎤 Speech ended");
      setIsRecording(false);
    };

    Voice.onSpeechResults = (e: SpeechResultsEvent) => {
      const transcript = e.value?.[0] || "";
      console.log("🎤 Speech results:", transcript);
      setTranscription(transcript);
    };

    Voice.onSpeechError = (e: SpeechErrorEvent) => {
      console.error("🎤 Speech error:", e);
      setError({
        message: "Speech recognition error. Please try again.",
        severity: "error",
      });
      setIsRecording(false);
    };

    // Cleanup
    return () => {
      Voice.destroy().then(Voice.removeAllListeners);
    };
  }, []);

  const startRecording = async () => {
    try {
      setError(null);
      setTranscription("");
      await Voice.start("en-US");
      Haptics.notificationAsync(Haptics.NotificationFeedbackType.Success);
    } catch (err) {
      console.error("🎤 Failed to start recording:", err);
      setError({
        message: "Failed to start recording",
        severity: "error",
      });
    }
  };

  const stopRecording = async () => {
    try {
      await Voice.stop();
      Haptics.notificationAsync(Haptics.NotificationFeedbackType.Success);
    } catch (err) {
      console.error("🎤 Failed to stop recording:", err);
      setError({
        message: "Failed to stop recording",
        severity: "error",
      });
    }
  };
```

**Reply by divyesh-puri** [reply]
Interesting! I am not sure if accuracy of STT will be better than deepgram though? Feel free to correct me

**Reply by MaheshtheDev** [reply]
it's been great so far, this will use native speech to text that Apple provides on device translation, not sure about the Android ( but for iOS its working pretty great so far)

**Comment by denieler** [comment]
hey 👋 I stumbled upon this question as well, when developing my react-native app and have managed to integrate Deepgram in my app. I've shared all the code here in the blog post https://denieler.com/blog/deepgram-integration-react-native. Hope it will help somebody to waste less time than I did 🙇‍♂️
